### ISSUE TITLE
records: isbn normalisation failing

### ISSUE THREAD

**Issue by lnielsen** [zenodo/zenodo#788]
```
ISBNError Root Cause
'Invalid ISBN format: 978-956-303-036- 5'
```

```
Stacktrace (most recent call last):

  File "flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "invenio_records_ui/views.py", line 182, in record_view
    return view_method(pid, record, template=template)
  File "invenio_records_ui/views.py", line 198, in default_view_method
    record=record,
  File "flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/zenodo/src/zenodo/zenodo/modules/records/templates/zenodo_records/record_detail.html", line 35, in top-level template code
    {%- set object_type = record.resource_type|objecttype %}
  File "/opt/zenodo/src/zenodo/zenodo/modules/theme/templates/zenodo_theme/page.html", line 25, in top-level template code
    {%- extends "invenio_theme/page.html" %}
  File "/opt/zenodo/lib/python2.7/site-packages/invenio_theme/templates/invenio_theme/page.html", line 73, in top-level template code
    {%- block body %}
  File "/opt/zenodo/lib/python2.7/site-packages/invenio_theme/templates/invenio_theme/page.html", line 79, in block "body"
    {%- block body_inner %}
  File "/opt/zenodo/lib/python2.7/site-packages/invenio_theme/templates/invenio_theme/page.html", line 83, in block "body_inner"
    {%- block page_body %}
  File "/opt/zenodo/src/zenodo/zenodo/modules/records/templates/zenodo_records/record_detail.html", line 91, in block "page_body"
    {%- include "zenodo_records/box/info.html" %}
  File "/opt/zenodo/src/zenodo/zenodo/modules/records/templates/zenodo_records/box/info.html", line 146, in top-level template code
    {%- set related_url = related_id.identifier|pid_url(scheme=related_id.scheme) %}{#  #}
  File "zenodo/modules/records/views.py", line 256, in pid_url
    return idutils.to_url(identifier, scheme)
  File "idutils/__init__.py", line 489, in to_url
    val = normalize_pid(val, scheme)
  File "idutils/__init__.py", line 481, in normalize_pid
    return normalize_isbn(val)
  File "idutils/__init__.py", line 448, in normalize_isbn
    return ISBN(val).hyphen()
  File "isbn/isbn.py", line 70, in __init__
    test = _normalize(str)
  File "isbn/isbn.py", line 24, in _normalize
    raise ISBNError("Invalid ISBN format: {}".format(str))
```

**Comment by slint** [comment]
Resolved through inveniosoftware/idutils#16, pending release.

**Comment by lnielsen** [comment]
https://github.com/inveniosoftware/idutils/pull/17
